invoices/invoiceregistry.go

```diff
@@ -846,7 +846,7 @@ func (i *InvoiceRegistry) notifyExitHopHtlcLocked(
 	}
 
 	if updateSubscribers {
-		i.notifyClients(ctx.hash, invoice, invoice.State)
+		defer i.notifyClients(ctx.hash, invoice, invoice.State)
```

Comment:
Can the timing of this call be more explicit? Defer for clean up is one thing, but using defer to run kick off notifications may be too much. Suggestion: don't return in the switch cases but do at at the very end of the function. Right before, notify clients.

Reply:
this is pulled out from a much larger pr which i'd rather not rebase for cosmetics. i guess we can just park this PR and circle back once those changes are in, it's not super urgent